status-handler.js

```diff
@@ -54,6 +56,32 @@ class StatusHandler {
   async event(data) {
     const { StatusEvent, StatusHead, Status } = this.models;
     const ev = this._transform(data, 'event');
+
+    // Fetch list of previous events for a certain build
+    const { pkg, version, env } = ev;
+    const previousEvents = await Status.findAll({
+      pkg,
+      version,
+      env
+    });
+
+    // Ensure that last event is the 'queued' event from carpentd
+    const lastEvent = previousEvents[previousEvents.length - 1];
+    if (lastEvent && lastEvent.message === 'Builds Queued') {
```

Comment:
id also confirm that `ev.message` is what you expect as well to help defend against data insertion raciness.